pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from pytorch_lightning.utilities.imports import (
-    _HOROVOD_AVAILABLE,
-    _IPU_AVAILABLE,
-    _TORCH_GREATER_EQUAL_1_8,
-    _TPU_AVAILABLE,
-)
-from pytorch_lightning.utilities.rank_zero import rank_zero_deprecation, rank_zero_info, rank_zero_warn
+from pytorch_lightning.utilities.imports import _HOROVOD_AVAILABLE, _IPU_AVAILABLE, _TPU_AVAILABLE
+
+log = logging.getLogger(__name__)
 
 if _HOROVOD_AVAILABLE:
     import horovod.torch as hvd
 
-log = logging.getLogger(__name__)
-
 
 class AcceleratorConnector:
     def __init__(
         self,
-        num_processes,
-        devices,
-        tpu_cores,
-        ipus,
-        accelerator,
-        strategy: Optional[Union[str, Strategy]],
-        gpus,
-        gpu_ids,
-        num_nodes,
-        sync_batchnorm,
-        benchmark,
-        replace_sampler_ddp,
-        deterministic: bool,
-        precision,
-        amp_type,
-        amp_level,
-        plugins,
-    ):
-        # initialization
-        self._device_type = _AcceleratorType.CPU
-        self._strategy_type = None
-        self._accelerator_type = None
-
-        self._strategy_flag = strategy.lower() if isinstance(strategy, str) else strategy
-        # TODO: Rename this to something else once all the distributed flags are moved to strategy
-        self.distributed_backend = accelerator
-
-        self._init_deterministic(deterministic)
-
-        self.num_processes = num_processes
-        self.devices = devices
-        # `gpus` is the input passed to the Trainer, whereas `gpu_ids` is a list of parsed gpu ids.
-        self.gpus = gpus
-        self.parallel_device_ids = gpu_ids
-        self.tpu_cores = tpu_cores
-        self.ipus = ipus
-        self.num_nodes = num_nodes
-        self.sync_batchnorm = sync_batchnorm
-        self.benchmark = benchmark
+        devices: Optional[Union[List[int], str, int]] = None,
+        num_nodes: int = 1,
+        accelerator: Optional[Union[str, Accelerator]] = None,
+        strategy: Optional[Union[str, Strategy]] = None,
+        plugins: Optional[Union[PLUGIN_INPUT, List[PLUGIN_INPUT]]] = None,
+        precision: Union[int, str] = 32,
+        amp_type: str = "native",
+        amp_level: Optional[str] = None,
+        sync_batchnorm: bool = False,
+        benchmark: bool = False,
+        replace_sampler_ddp: bool = True,
+        deterministic: bool = False,  # TODO: why is it unused?
+        num_processes: Optional[int] = None,  # deprecated
+        tpu_cores: Optional[Union[List[int], int]] = None,  # deprecated
+        ipus: Optional[int] = None,  # deprecated
+        gpus: Optional[Union[List[int], str, int]] = None,  # deprecated
+        gpu_ids: Optional[List[int]] = None,  # TODO: why is it unused?
+    ) -> None:
+        """The AcceleratorConnector parses several Trainer arguments and instantiates the Strategy including other
```

Comment:
Let's add more meat to the docstrings, following the code is pretty complex, and more comments would be helpful.

Reply:
Yeah definitely. Added to the follow up items in #11449 Then we can work on it parallelly, otherwise this PR is the blocker for other 1.6 effort